chore(docs): monorepos support (#36504)
gatsbyjs/gatsby: docs/docs/reference/cloud/monorepos.md

---
title: "Monorepos"
description: "Monorepos support in Gatsby Cloud"
---

## Introduction

> Gatsby Cloud has support for monorepos projects, but since there is a plethora of dependency management tools out there some issues can occure.
> This document will walk you through the tools that are supported and how to troubleshoot some edge cases

Reviewer: ```suggestion
Gatsby Cloud has support for projects organized as [monorepos](https://monorepo.tools/), but since there are different competing tools to build a monorepo it's important to know what Gatsby Cloud supports today.

This document will explain which tools are supported and how you can troubleshoot issues.
```

> ## Monorepo Support

Reviewer: ```suggestion
## Supported tools
```

> The following tables shows the level of support for each tool

Reviewer: ```suggestion
The following table shows the level of support for each tool.
```

**Legend**

| Icon | Feature Capability                       |
| ---- | ---------------------------------------- |
| ●    | Fully Supported                          |
| ◐    | Somewhat Supported (support is minimal ) |
| ○    | Not Supported                            |

**Support**

| Tool.                            | Level of Support | Notes                                                   |
| -------------------------------- | ---------------- | ------------------------------------------------------- |
| Yarn Workspaces (v1)             | ◐                |                                                         |

> | Yarn Workspaces (v2/v3 with PnP) | ◐                |                                                         |

Reviewer: Why is it only "somewhat supported", what is missing?

Author: we identified a bug there and it's being worked on on the cloud side

Reviewer 2: is V1 borked?

Author: nope only with pnp, I will change that

| Lerna                            | ●                |                                                         |
| NPM Workspaces (v7 and onwards)  | ●                |                                                         |

> | NX                               | ○                | Work is in progress                                     |

Reviewer: ```suggestion
| NX                               | ○                | Work in progress                                     |
```

| Turborepo                        | ○                | We do not have plans to support Turborepo at this time. |

> ## Troubleshooting

Reviewer: Completely missing?

Author: The only thing I can put here for now is redirecting people to support when they have issues and then we can actively add more stuff to it when we discover any edge case

Reviewer 2: Can we add something around pnp? If you're having issues use `nodeLinker: node-modules`

https://yarnpkg.com/features/pnp/#pnp-loose-mode
https://yarnpkg.com/getting-started/migration/#if-required-enable-the-node-modules-plugin
https://yarnpkg.com/getting-started/recipes/#hybrid-pnp--node_modules-mono-repo

Reviewer 2: Can we add something around pnp? If you're having issues use `nodeLinker: node-modules`

https://yarnpkg.com/features/pnp/#pnp-loose-mode
https://yarnpkg.com/getting-started/migration/#if-required-enable-the-node-modules-plugin
https://yarnpkg.com/getting-started/recipes/#hybrid-pnp--node_modules-mono-repo
